### PR TITLE
Core/Movement: move MoveSplineInit

### PR DESCRIPTION
**Pull request by ccrs**
move spline initializers and support for MovementInform

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 1]
```diff
@@ -382,12 +382,10 @@ class npc_eye_of_acherus : public CreatureScript
 
                 me->SetReactState(REACT_PASSIVE);
 
-                me->GetMotionMaster()->MovePoint(POINT_EYE_FALL, EyeOFAcherusFallPoint, false);
-
                 Movement::MoveSplineInit init(me);
                 init.MoveTo(EyeOFAcherusFallPoint.GetPositionX(), EyeOFAcherusFallPoint.GetPositionY(), EyeOFAcherusFallPoint.GetPositionZ(), false);
                 init.SetFall();
-                init.Launch();
+                me->GetMotionMaster()->LaunchMoveSpline(std::move(init), POINT_EYE_FALL, MOTION_SLOT_ACTIVE);
```
Comment (ccrs) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: @Shauren before they had to use movepoint and spline to have movementinform + "custom args"

Comment (ccrs) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: there are only two cases of this though

Comment (Shauren) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: Doesn't `MovementInform` hook need to be updated as well to check EFFECT_MOTION_TYPE insted of POINT_MOTION_TYPE?

File: src/server/game/Movement/MovementGenerators/PointMovementGenerator.h [hunk 2]
```diff
@@ -64,9 +66,10 @@ class AssistanceMovementGenerator : public PointMovementGenerator<Creature>
 class EffectMovementGenerator : public MovementGenerator
 {
     public:
-        explicit EffectMovementGenerator(uint32 id) : _pointId(id) { }
+        explicit EffectMovementGenerator(uint32 id) : _pointId(id), _duration(0) { }
```
Comment (Shauren) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.h]: Is this constructor used anywhere after this PR? If not then you can drop it (together with Optional use)

File: src/server/game/Movement/Spline/MoveSplineInit.h [hunk 3]
```diff
@@ -53,6 +53,11 @@ namespace Movement
     public:
 
         explicit MoveSplineInit(Unit* m);
+        MoveSplineInit(MoveSplineInit&& init) : args(std::move(init.args))
```
Comment (Shauren) [inline src/server/game/Movement/Spline/MoveSplineInit.h]: pretty sure you can just `MoveSplineInit(MoveSplineInit&& init) = default` here
also, you need to do the same for MoveSplineInitArgs (default one is not generated by compiler if you have "user-declared" destructors)

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp (hunk 1 above)
Comment (ccrs) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: true true

File: src/server/game/Movement/Spline/MoveSplineInit.h (hunk 3 above)
Comment (ccrs) [reply src/server/game/Movement/Spline/MoveSplineInit.h]: daaaaaaaamn

File: src/server/game/Movement/MovementGenerators/PointMovementGenerator.h (hunk 2 above)
Comment (ccrs) [reply src/server/game/Movement/MovementGenerators/PointMovementGenerator.h]: I think I'll also move it to another file and rename

File: src/server/game/Movement/MovementGenerators/GenericMovementGenerator.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+#include "GenericMovementGenerator.h"
+#include "Creature.h"
+#include "CreatureAI.h"
+#include "MoveSpline.h"
+#include "MoveSplineInit.h"
+#include "Unit.h"
+
+void GenericMovementGenerator::Initialize(Unit*/* owner*/)
```
Comment (illfated) [inline src/server/game/Movement/MovementGenerators/GenericMovementGenerator.cpp]: Bad comment format - VS debug error.

File: src/server/game/Movement/MotionMaster.cpp [hunk 5]
```diff
@@ -253,12 +253,8 @@ void MotionMaster::MoveFollow(Unit* target, float dist, ChaseAngle angle, Moveme
     if (!target || target == _owner)
         return;
 
-    TC_LOG_DEBUG("misc", "%s (Entry: %u, GUID: %u) follows %s (Entry %u, GUID: %u).",
-        _owner->GetTypeId() == TYPEID_PLAYER ? "Player" : "Creature",
-        _owner->GetEntry(), _owner->GetGUID().GetCounter(),
-        target->GetTypeId() == TYPEID_PLAYER ? "player" : "creature",
-        target->GetEntry(), target->GetGUID().GetCounter());
-
+    TC_LOG_DEBUG("misc", "MotionMaster::MoveFollow: %s (Entry: %u, GUID: %u) follows %s (Entry %u, GUID: %u)", _owner->GetTypeId() == TYPEID_PLAYER ? "Player" : "Creature", _owner->GetEntry(), _owner->GetGUID().GetCounter(),
```
Comment (joschiwald) [inline src/server/game/Movement/MotionMaster.cpp]: ObjectGuid::ToString can simplify this log message

Comment (ccrs) [reply src/server/game/Movement/MotionMaster.cpp]: just did some formating no real change, but I can give it a look

Comment (Shauren) [reply src/server/game/Movement/MotionMaster.cpp]: I really want people to use ObjectGuid::ToString instead of manually trying to print its pieces...

**Comment by Killyana** [comment]
I cannot use charge in transport, Npcs are kind of flying on the gunship and http://www.wowhead.com/npc=36978/skybreaker-mortar-soldier are not casting any rockets, don't know if it's related

**Comment by ccrs** [comment]
test now @Killyana

**Comment by Killyana** [comment]
Vehicle issue fixed, other movements issue on transports looks to be related to something else.

**Comment by ccrs** [comment]
yeah, they are not related, same as captains on gunship battle flying around

**Comment by Aokromes** [comment]
Maybe captain thingy is because https://github.com/TrinityCore/TrinityCore/issues/21872

**Comment by ccrs** [comment]
most likely

**Comment by jackpoz** [comment]
this is the cause of https://github.com/TrinityCore/TrinityCore/issues/22859

**Pull request merged**